tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -420,6 +420,139 @@ def verify_failure_received():
         finally:
             lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
 
+    @pytest.mark.parametrize(
+        "item_to_put1, item_to_put2, filter, calls",
+        [
+            # Test with no filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                None,
+                2,
+            ),
+            # Test with filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"]},
+                1,
+            ),
+            # Test with OR filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT", "MODIFY"]},
+                2,
+            ),
+            # Test with 2 filters (AND), and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"], "eventSource": ["aws:dynamodb"]},
+                1,
+            ),
+            # Test exists filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
+                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+                1,
+            ),
+            # numeric filters
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "123"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">", 100]}]}}},
+                1,
+            ),
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">=", 100, "<", 200]}]}}},
+                1,
+            ),
+            # Prefix
+            (
+                {"id": {"S": "test123"}, "prefix": {"S": "us-1-testtest"}},
+                {"id": {"S": "test1234"}, "prefix": {"S": "testtest"}},
+                {"dynamodb": {"NewImage": {"prefix": [{"prefix": "us-1"}]}}},
+                1,
+            ),
+        ],
+    )
+    def test_dynamo_db_insert_event(
+        self,
+        create_lambda_function,
+        lambda_client,
+        dynamodb_client,
+        dynamodb_create_table,
```

Comment:
please add the following fixtures here:
```
        create_role,
        create_iam_role_with_policy,
        logs_client,
        wait_for_dynamodb_stream_ready,
```

`wait_for_dynamodb_stream_ready` is a new fixture that should ensure that the stream is ready to process. You can add it to `fixtures.py`, something like this would suffice:
```
@pytest.fixture
def wait_for_dynamodb_stream_ready(dynamodbstreams_client):
    def _wait_for_stream_ready(stream_arn: str):
        def is_stream_ready():
            describe_stream_response = dynamodbstreams_client.describe_stream(StreamArn=stream_arn)
            return describe_stream_response["StreamDescription"]["StreamStatus"] == "ENABLED"

        poll_condition(is_stream_ready)

    return _wait_for_stream_ready
```